Libraries/Components/MaskedView/MaskedViewIOS.ios.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule MaskedViewIOS
+ * @flow
+ */
+
+const React = require('React');
+const StyleSheet = require('StyleSheet');
+const View = require('View');
+const ViewPropTypes = require('ViewPropTypes');
+const findNodeHandle = require('ReactNative').findNodeHandle;
+const requireNativeComponent = require('requireNativeComponent');
+const cloneReferencedElement = require('react-clone-referenced-element');
+const ReactPropTypes = React.PropTypes;
+
+import type { ViewProps } from 'ViewPropTypes';
+
+type Props = ViewProps & {
+  /**
+   * Only a single child is supported.
+   */
+  children: React.Element<*>,
+  /**
+   * Should return a React element to be rendered and applied as the
+   * mask for the child element.
+   */
+  renderMask: () => React.Element<*>,
```

Comment:
Are there cases where MaskedView may have information that affects the rendering of the mask view that would make sense to pass in as args to the render function? (This is the rationale for renderScrollComponent() for composing scrollable components.)